apps/blockchain/lib/blockchain/chain.ex

```diff
@@ -161,6 +164,32 @@ defmodule Blockchain.Chain do
     end
   end
 
+  @doc """
+  Get the EVM configuration based on the chain and block number
+  """
+  def evm_config(chain = %__MODULE__{}, block_number \\ nil) do
```

Comment:
I didn't get the idea. shouldn't all methods in configurations now accept block_number? or block_number will be stored in the configuration before passing it to evm?

Reply:
The correct EVM configuration [is chosen](https://github.com/poanetwork/mana/blob/mf-use-correct-evm-config/apps/blockchain/lib/blockchain/block.ex#L725) before processing a transaction based on block number. The idea being that our configuration won't change mid-transaction.

When processing a transaction any call to a configuration method will call the correct fork module `Configuration.Homestead`/ `Configuration.Frontier` etc.